components/ai_chat/resources/page/components/main/index.tsx

```diff
@@ -4,21 +4,93 @@
  * You can obtain one at https://mozilla.org/MPL/2.0/. */
 
 import * as React from 'react'
-import styles from './style.module.scss'
 import Icon from '@brave/leo/react/icon'
 import Button from '@brave/leo/react/button'
 
-interface MainProps {
-  conversationList: React.ReactNode
-  inputBox: React.ReactNode
-  siteTitle?: React.ReactNode
-  promptAutoSuggestion?: React.ReactNode
-  currentErrorElement?: React.ReactNode
-  onSettingsClick?: () => void
-  onEraseClick?: () => void
-}
+import styles from './style.module.scss'
+import ConversationList from '../conversation_list'
+import PrivacyMessage from '../privacy_message'
+import SiteTitle from '../site_title'
+import PromptAutoSuggestion from '../prompt_auto_suggestion'
+import ErrorConnection from '../error_connection'
+import ErrorRateLimit from '../error_rate_limit'
+import InputBox from '../input_box'
+import getPageHandlerInstance, { AutoGenerateQuestionsPref, APIError } from '../../api/page_handler'
+import DataContext from '../../state/context'
+
+function Main () {
+  const { siteInfo, userAutoGeneratePref, setUserAllowsAutoGenerating, generateSuggestedQuestions,
+    hasSeenAgreement, currentError } = React.useContext(DataContext)
+
+  const apiHasError = !!currentError && (currentError !== APIError.None)
+
+  const handleQuestionSubmit = (question: string) => {
+    getPageHandlerInstance().pageHandler.submitHumanConversationEntry(question)
+  }
+
+  const handlePageContentDisconnect = () => {
+    getPageHandlerInstance().pageHandler.disconnectPageContents()
+  }
+
+  const handleOnEnableAutoGenerateQuestion = () => {
+    setUserAllowsAutoGenerating(true)
+    generateSuggestedQuestions()
+  }
+
+  const handleSettingsClick = () => {
+    getPageHandlerInstance().pageHandler.openBraveLeoSettings()
+  }
+
+  const handleEraseClick = () => {
+    getPageHandlerInstance().pageHandler.clearConversationHistory()
+  }
```

Comment:
why not put all this on the context too?

Reply:
most of these are used in one place and not many. for ex, handleSettings/EraseClick are just part of header and this file only.

Reply:
However, I have moved some functions into it's component file as we don't need prop drilling for those anymore. It should be cleaned for now.